importers/wecdis_importer.py

```diff
@@ -0,0 +1,493 @@
+            elif msg_type == MsgType.TMA:
+                self.handle_tma(data_store, line_number, tokens, datafile, change_id)
+            elif MsgType.TTM in msg_type:
+                self.handle_ttm(data_store, line_number, tokens, datafile, change_id)
+        datafile.flush_extracted_tokens()
+
+    def handle_vnm(self, vnm_tokens, line_number):
+        """Extracts the important information from a VNM line
+        :param vnm_tokens: A tokenised VNM line
+        :ptype vnm_tokens: Line (list of tokens)"""
+        if len(vnm_tokens) < 3:
+            self.errors.append(
+                {self.error_type: f"Not enough parts in line {line_number}. No platform name found"}
+            )
+            return  # Not necessarily an error, but can't do anything with it now
+        if vnm_tokens[1].text != "VNM":
+            # Programming error, shouldn't be hit but if it does we're passing the wrong line
+            raise TypeError(f"Expected a VNM line, given {vnm_tokens[1].text}")
+        # Ignore the *XX part if present
+        platform_name_token = vnm_tokens[2]
+        self.platform_name, _, _ = platform_name_token.text.partition("*")
+        platform_name_token.record(self.name, "platform name", self.platform_name)
+
+    def handle_depth(self, tokens, line_number):
+        """Extracts the depth from the PDS line
+        :param tokens: A tokenised PDS line
+        :ptype tokens: Line (list of tokens)"""
+        depth_token = tokens[2]
+        depth_units_token = tokens[3]  # This is an assumption about the unit being position 3
+        if depth_token and depth_token.text:
+            # Making an assumption of how this would look if using FT.
+            # [3] may not be the units but unclear from a single sample file
+            unit_text, _, _ = depth_units_token.text.upper().partition("*")
+            units = unit_registry.foot if unit_text == "FT" else unit_registry.meter
+            depth_valid, depth = convert_distance(
+                depth_token.text, units, line_number, self.errors, self.error_type
+            )
+            if depth_valid:
+                self.elevation = -1 * depth
+            depth_token.record(self.name, "Depth", self.elevation)
+
+    def handle_timestamp(self, dza_tokens, line_number):
+        """Extracts the important information from a DZA (Timestamp) line
+        :param dza_tokens: A tokenised DZA line
+        :ptype dza_tokens: Line (list of tokens)"""
+        if len(dza_tokens) < 4:
+            self.errors.append(
+                {self.error_type: f"Not enough parts in line {line_number}. No timestamp provided."}
+            )
+            return
+        if dza_tokens[1].text != "DZA":
+            raise TypeError(f"Expected a DZA line, given {dza_tokens[1].text}")
+        date = dza_tokens[2].text
+        time = dza_tokens[3].text
+        # TODO - confirm where the timezone is determined (presumably not always GMT)
```

Comment:
I see this TODO about time zone is still in here: any progress on this?

Reply:
Onsite I only found two WECDIS files.  One had a `TZS` entry with `UTC`, the other didn't have a TZS entry.

I'll send a message to try to get more sample data.